flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/sinks/TableSinkUtils.scala

```diff
@@ -189,13 +190,45 @@ object TableSinkUtils {
       queryLogicalType,
       withChangeFlag)
     if (LogicalTypeChecks.isCompositeType(requestedOutputType.getLogicalType)) {
-      DataTypeUtils.expandCompositeTypeToSchema(requestedOutputType)
+      fromDataTypeToTypeInfo(requestedOutputType) match {
```

Comment:
This is just for RawType? Can you add comment?

Reply:
This is for Pojo type. We have to compare query schema with POJO sink schema by field names instead of field indexes.

Reply:
I will add a comment.